controllers/argocd/applicationset.go

```diff
@@ -58,6 +62,20 @@ func getArgoApplicationSetCommand(cr *argoproj.ArgoCD) []string {
 		cmd = append(cmd, ApplicationSetGitlabSCMTlsCertPath)
 	}
 
+	if len(cr.Spec.ApplicationSet.SourceNamespaces) > 0 {
+		cmd = append(cmd, "--applicationset-namespaces", fmt.Sprint(strings.Join(cr.Spec.ApplicationSet.SourceNamespaces, ",")))
+	}
```

Comment:
@svghadi at this point should we also compare this list against the app source namespace list to make sure the appset list is a proper subset of the app source list? if we find any namespaces that are not listed in the app source list we should log it and skip it

Reply:
done